src/go/rdctl/cmd/shell.go

```diff
@@ -0,0 +1,129 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+)
+
+// shellCmd represents the shell command
+var shellCmd = &cobra.Command{
+	Use:   "shell",
+	Short: "Run an interactive shell or a command in a Rancher Desktop-managed VM",
+	Long: `Run an interactive shell or a command in a Rancher Desktop-managed VM. For example:
+
+> rdctl shell
+-- Runs an interactive shell
+> rdctl shell ls -CF /tmp
+-- Runs 'ls -CF' from /tmp on the VM
+> rdctl shell bash -c "cd .. ; pwd"
+-- Usual way of running multiple statements on a single call
+`,
+	DisableFlagParsing: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Do manual flag parsing looking to see if we should give help instead.
+		if len(args) > 0 && (args[0] == "-h" || args[0] == "--help") {
+			return cmd.Help()
+		}
+		return doShellCommand(cmd, args)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(shellCmd)
+}
+
+func doShellCommand(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+	var commandName string
+	if runtime.GOOS == "windows" {
+		commandName = "wsl"
+		args = append([]string{"-d", "rancher-desktop"}, args...)
+	} else {
+		err := addLimaBinToPath()
+		if err != nil {
+			return err
+		}
+		err = setupLimaHome()
+		if err != nil {
+			return err
+		}
+		commandName = "limactl"
+		args = append([]string{"shell", "0"}, args...)
+	}
+	shellCommand := exec.Command(commandName, args...)
+	shellCommand.Stdin = os.Stdin
+	shellCommand.Stdout = os.Stdout
+	shellCommand.Stderr = os.Stderr
+	return shellCommand.Run()
+}
+
+func addLimaBinToPath() error {
+	_, err := exec.LookPath("limactl")
+	if err == nil {
+		// It's already in the pth
+		return err
+	}
+	execPath, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	execPath, err = filepath.EvalSymlinks(execPath)
+	if err != nil {
+		return err
+	}
+	candidatePath := path.Join(path.Dir(path.Dir(execPath)), "lima", "bin")
+	notFoundError := fmt.Errorf("no executable limactl file found in %s; try rerunning with the directory containing `limactl` added to PATH", candidatePath)
+	stat, err := os.Stat(path.Join(candidatePath, "limactl"))
+	if err != nil {
+		return notFoundError
+	}
+	if uint32(stat.Mode().Perm())&0111 == 0 {
+		return notFoundError
+	}
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", candidatePath, os.Getenv("PATH")))
+	return nil
+}
+
+func setupLimaHome() error {
+	if os.Getenv("LIMA_HOME") != "" {
+		// It's already in the environment
+		return nil
+	}
```

Comment:
Again, we don't want this. We only ever want to use the Rancher Desktop VM. Checking the environment will break `rdctl shell` because it will point to the user's `LIMA_HOME`, which will not contain the `0` Rancher Desktop instance.